src/chatd.cpp

```diff
@@ -744,7 +749,8 @@ HistSource Chat::getHistoryFromDbOrServer(unsigned count)
 
 void Chat::requestHistoryFromServer(int32_t count)
 {
-    assert(mConnection.isOnline());
+    // the connection must be established (for a JOIN + HIST), but not yet logged in
+    assert(mConnection.state() >= Connection::kStateConnected);
```

Comment:
To match the comment, the assert should be `mConnection.state() == Connection::kStateConnected`